pkg/cmd/build/translator.go

```diff
@@ -0,0 +1,267 @@
+		userNs:          okCtx.GetNamespace(),
+		globalNamespace: globalNamespace,
+	}
+}
+
+func (rt registryTranslator) translate(line string) string {
+	if strings.Contains(line, constants.DevRegistry) {
+		result := rt.replacer.Replace(line, constants.DevRegistry, rt.userNs)
+		return result
+	}
+
+	if strings.Contains(line, constants.GlobalRegistry) {
+		result := rt.replacer.Replace(line, constants.GlobalRegistry, rt.globalNamespace)
+		return result
+	}
+	return line
+}
+
+type cacheMountTranslator struct {
+	repo               string
+	dockerfilePath     string
+	buildTarget        string
+	hash               func(repositoryURL, dockerfilePath, buildTarget, cacheTarget string) string
+	cacheMountRegex    *regexp.Regexp
+	hasIDRegex         *regexp.Regexp
+	targetExtractRegex *regexp.Regexp
+	mountRegex         *regexp.Regexp
+	targetRegex        *regexp.Regexp
+}
+
+func newCacheMountTranslator(repo, dockerfilePath, target string) cacheMountTranslator {
+
+	return cacheMountTranslator{
+		repo:               repo,
+		dockerfilePath:     dockerfilePath,
+		buildTarget:        target,
+		hash:               generateProjectHash,
+		cacheMountRegex:    cacheMountRegex,
+		hasIDRegex:         hasIDRegex,
+		targetExtractRegex: targetExtractRegex,
+		mountRegex:         mountRegex,
+		targetRegex:        targetRegex,
+	}
+}
+
+func generateProjectHash(repositoryURL, dockerfilePath, buildTarget, cacheTarget string) string {
+	// Create input string for hashing
+	input := fmt.Sprintf("build-%s-%s-%s-%s", repositoryURL, dockerfilePath, buildTarget, cacheTarget)
+
+	// Generate SHA256 hash
+	hasher := sha256.New()
+	hasher.Write([]byte(input))
+	hash := hasher.Sum(nil)
+
+	// Return first 12 characters of hex hash for readability
+	return hex.EncodeToString(hash)[:12]
+}
+
+func (cmt cacheMountTranslator) translate(line string) string {
+
+	// Check if this RUN command has a cache mount
+	if !cmt.cacheMountRegex.MatchString(line) {
+		return line
+	}
+
+	// Find all --mount= occurrences and process them individually
+	result := line
+
+	// Find all mount definitions
+	matches := cmt.mountRegex.FindAllStringSubmatch(result, -1)
+	for _, match := range matches {
+
+		// Check that the mountID has mount and params
+		// Example: --mount=id=test,type=cache,target=...
+		// [0] --mount=id=test,type=cache,target=...
+		// [1] id=test,type=cache,target=...
+		if len(match) < 2 {
+			continue
+		}
+
+		fullMount := match[0]   // --mount=type=cache,target=...
+		mountParams := match[1] // type=cache,target=...
+
+		// Check if this is a cache mount
+		// Example: type=cache,target=... => true
+		// Example: type=secret => false
+		if !strings.Contains(mountParams, "type=cache") {
+			continue
+		}
+
+		// Check if it already has an id
+		// Example: id=test,type=cache,target=... => true
+		// Example: type=cache,target=... => false
+		if strings.Contains(mountParams, "id=") {
```

Comment:
This check is wrong. if the mount instruction uses `uid=` or `guid=`, this would be true and it would not add the id. Example of instruction: 

```
RUN --mount=type=cache,target=/home/okteto/.npm,uid=1001,gid=1001 \
    npm ci --cache /home/okteto/.npm
```